### ISSUE TITLE
Support for Docusaurus 3.1.1

### ISSUE THREAD

**Issue by timolagus** [typesense/docusaurus-theme-search-typesense#42]
## Description

Add support for Docusaurus 3.1.1.

## Steps to reproduce

Update to the following dependency versions in `package.json`:

```
@docusaurus/core: 3.1.1
@docusaurus/module-type-aliases: 3.1.1
@docusaurus/preset-classic: 3.1.1
@docusaurus/theme-live-codeblock: ^3.1.1
@docusaurus/theme-mermaid: ^3.1.1
@docusaurus/types: 3.1.1
@mdx-js/react: ^3.0.1
```

## Expected Behavior

`npm install` completes without issue.

## Actual Behavior

`npm install` fails with:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: docusaurus-theme-search-typesense@0.15.0
npm ERR! Found: @docusaurus/core@3.1.1
npm ERR! node_modules/@docusaurus/core
npm ERR!   @docusaurus/core@"3.1.1" from the root project
npm ERR!   @docusaurus/core@"3.1.1" from @docusaurus/plugin-debug@3.1.1
npm ERR!   node_modules/@docusaurus/plugin-debug
npm ERR!     @docusaurus/plugin-debug@"3.1.1" from @docusaurus/preset-classic@3.1.1
npm ERR!     node_modules/@docusaurus/preset-classic
npm ERR!       @docusaurus/preset-classic@"3.1.1" from the root project
npm ERR!   24 more (@docusaurus/plugin-google-analytics, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @docusaurus/core@"3.1.0" from docusaurus-theme-search-typesense@0.15.0
npm ERR! node_modules/docusaurus-theme-search-typesense
npm ERR!   docusaurus-theme-search-typesense@"^0.15.0" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: @docusaurus/core@3.1.0
npm ERR! node_modules/@docusaurus/core
npm ERR!   peer @docusaurus/core@"3.1.0" from docusaurus-theme-search-typesense@0.15.0
npm ERR!   node_modules/docusaurus-theme-search-typesense
npm ERR!     docusaurus-theme-search-typesense@"^0.15.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

## Metadata

**Typesense Version**: 0.25.1

**OS**: Ubuntu 22.04.3 LTS

**Comment by jasonbosco** [comment]
Just published `0.16.0-0` of the theme. Could you try it with that version?

**Comment by jasonbosco** [comment]
Thank you for confirming! I'll publish 0.16.0 shortly.
